File: app/src/test/java/br/com/iomarsantos/testeandroid/ui/utils/rx/TestSchedulerProvider.java
package br.com.iomarsantos.testeandroid.ui.utils.rx;

import br.com.iomarsantos.testeandroid.utils.rx.SchedulerProvider;
import io.reactivex.Scheduler;
import io.reactivex.schedulers.TestScheduler;

public class TestSchedulerProvider implements SchedulerProvider {

    private final TestScheduler mTestScheduler;

    public TestSchedulerProvider(TestScheduler testScheduler) {
        this.mTestScheduler = testScheduler;
    }

    @Override
    public Scheduler ui() {
        return mTestScheduler;
    }

    @Override
    public Scheduler computation() {
        return mTestScheduler;
    }

    @Override
    public Scheduler io() {
        return mTestScheduler;
    }

}

File: SantanderInvestimento/app/src/test/java/br/com/santander/santanderinvestimento/util/TestSchedulerProvider.kt
package br.com.santander.santanderinvestimento.util


import br.com.santander.santanderinvestimento.util.rx.SchedulerProvider
import io.reactivex.schedulers.Schedulers

class TestSchedulerProvider : SchedulerProvider {
    override fun io() = Schedulers.trampoline()

    override fun ui() = Schedulers.trampoline()

}